Matrix.ConsoleApp/MatrixExtension.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq;
+
+namespace Matrix.ConsoleApp
+{
+    internal static class MatrixExtension 
+    {
+        public static Matrix GetNewMatrixFromAddingTwoMatrices(this Matrix first, Matrix second)
+        {
+            //adding elements of two arrays
```

Comment:
Simplify, do not repeat yourself, combine lines from 11 till 27 in one block, not reassign  memory in 22. Determine the needed memory size before line 11 and not do the operation twice